tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/AMQPFederationConnectTest.java

```diff
@@ -219,6 +219,7 @@ public void testFederationCreatesControlLinkAndClosesConnectionDetachIndicatesNo
                                        .withNullTarget();
          peer.remoteDetach().withErrorCondition("amqp:unauthorized-access", "Not authroized").queue();
          peer.expectDetach().optional();
+         peer.expectClose().optional();
```

Comment:
Moving to the Connection Failure to the Netty thread unveiled an issue with Broker Connection. By the time I issue a failure, we won't wait the flush of the connection.close to be sent to the client before we close it... we just disconnect(). If you are writing a unit test like this.. the expect close becomes optional. Fixing it would require some work, and it would be orthogonal to this.

As for now, this is what I"m supposed to do with these tests.

Reply:
Cant say I agree with that. Doesnt seem at all orthogonal for this change to be stopping it doing what it was/should.

Reply:
If you ran the test Long enough before it would probably also fail.

all I did was to move the connectionFailure to the correct thread.

Reply:
Just quick note to say that it wasn't sending close before, and only occasionally sending the detach.  It seems now it sometimes sends both, and sometimes not.  Would be nice if it sent them both as a well behaved peer should, but for clarity the original test omitted this as it was never found to have been sent.